cms/static/js/views/modals/course_outline_modals.js

```diff
@@ -300,35 +307,39 @@ define(['jquery', 'backbone', 'underscore', 'gettext', 'js/views/baseview',
                 'maxTime': '05:00',
                 'forceRoundTime': false
             });
+
+            this.setExamType(this.model.get('is_time_limited'), this.model.get('is_proctored_enabled'),
+                            this.model.get('is_practice_exam'));
             this.setExamTime(this.model.get('default_time_limit_minutes'));
-            this.setExamTmePreference(this.model.get('is_time_limited'));
-            this.setExamProctoring(this.model.get('is_proctored_enabled'));
-            this.setPracticeExam(this.model.get('is_practice_exam'));
-        },
-        setPracticeExam: function(value) {
-            this.$('#id_practice_exam').prop('checked', value);
         },
-        setExamProctoring: function(value) {
-            this.$('#id_exam_proctoring').prop('checked', value);
+        setExamType: function(is_time_limited, is_proctored_enabled, is_practice_exam) {
+            if (!is_time_limited) {
+                this.$("#id_not_timed").prop('checked', true);
+                return;
+            }
+
+            if (this.options.enable_proctored_exams) {
+                this.$('#id_time_limit_div').show();
+
+                if (is_proctored_enabled) {
+                    if (is_practice_exam) {
+                        this.$('#id_practice_exam').prop('checked', true);
+                    } else {
+                        this.$('#id_proctored_exam').prop('checked', true);
+                    }
+
+                } else {
+                    this.$("#id_timed_exam").prop('checked', true);
```

Comment:
I don't understand why it is assumed that this is a timed exam. Can add a comment explaining the logic?